Automattic/wp-calypso
# Anchor: Fix back error with podcast title

#### Changes proposed in this Pull Request

* `useSiteTitle` felt like an unnecessary abstraction, so remove this hook and move the logic directly into the podcast title step for now.
* Instead of saving the API podcast title to the onboarding store, simply return it from the API.
* If the site title is present in the onboarding store, ignore the podcast title from the API. If not, auto-fill with the title from the API.
* This prevents the API from overwriting a custom site title in the store when we use the Back button to navigate back to the podcast title step.

#### Testing instructions

* Apply D80216-code to your sandbox and sandbox `public-api.wordpress.com`
* Switch to this PR, navigate to `/setup?anchor_podcast=[your podcast ID]`
* You should see the form title auto-filled with your podcast title
* Change the title to something new
* Go to the next step
* Click Back
* The podcast title should still be your custom value from step 3

Related to #63212
